build.py

```diff
@@ -23,18 +27,263 @@ def wait_until_container_is_ready():
+    if(compare_results(result,r'^'+dc)):
+        return True
+    return False
+
+def get_dc_def(openshift_url,microserver_url,project,docker_image):
+    dc=""
+    if(openshift_url is not None and len(openshift_url)>0 and
+       microserver_url is not None and len(microserver_url)>0 and
+       project is not None and len(project)>0 and
+       docker_image is not None and len(docker_image)>0):
+        dc='apiVersion: v1\n' \
+        +'kind: DeploymentConfig\n' \
+        +'metadata: \n' \
+        +'  name: '+ project +'\n' \
+        +'  labels:\n' \
+        +'    app: '+ project +'\n' \
+        +'spec:\n' \
+        +'  replicas: 1\n' \
+        +'  selector:\n' \
+        +'    app: '+project+'\n' \
+        +'    deploymentconfig: '+project+'\n' \
+        +'  strategy:\n' \
+        +'    activeDeadlineSeconds: 21600\n' \
+        +'    rollingParams:\n' \
+        +'      intervalSeconds: 1\n' \
+        +'      maxSurge: 25%\n' \
+        +'      maxUnavailable: 25%\n' \
+        +'      timeoutSeconds: 600\n' \
+        +'      updatePeriodSeconds: 1\n' \
+        +'    type: Rolling\n' \
+        +'  template:\n' \
+        +'    metadata:\n' \
+        +'      labels:\n' \
+        +'        app: '+project+'\n' \
+        +'        deploymentconfig: '+project+'\n' \
+        +'    spec:\n' \
+        +'      serviceAccountName: '+project+'-sa\n' \
+        +'      automountServiceAccountToken: True\n' \
+        +'      containers:\n' \
+        +'      - env:\n' \
+        +'        - name: openshift_url\n' \
+        +'          value: '+openshift_url+'\n' \
+        +'        image: '+docker_image+'\n' \
+        +'        imagePullPolicy: Always\n' \
+        +'        name: '+project+'\n' \
+        +'        resources:\n' \
+        +'          limits:\n' \
+        +'            memory: 1024Mi\n' \
+        +'            cpu: 2000m\n' \
+        +'          requests:\n' \
+        +'            memory: 150Mi\n' \
+        +'            cpu: 250m\n' \
+        +'        #terminationMessagePath: /dev/termination-log\n' \
+        +'      dnsPolicy: ClusterFirst\n' \
+        +'      restartPolicy: Always\n' \
+        +'      terminationGracePeriodSeconds: 30\n' \
+        +'    paramters:\n' \
+        +'    - name: "openshifturl"\n' \
+        +'      displayName: "OpenShift URL"\n' \
+        +'      description: "The OpenShift Master URL - because openshift pods don\'t know their master"\n' \
+        +'      required: true"\n' \
+        +'      value: ' + openshift_url + '\n' \
+        +'  test: False\n' \
+        +'  triggers:\n' \
+        +'  - type: ConfigChange\n'
+    return dc
+
+def oc_create_dc(openshift_url,microserver_url,project,docker_image):
+    proc=subprocess.Popen(['oc','create','-f','-'],stdout=subprocess.PIPE,stderr=subprocess.STDOUT,stdin=subprocess.PIPE)
+    proc.communicate(get_dc_def(openshift_url,microserver_url,project,docker_image).encode())
+    return 1
```

Comment:
why `return 1`?

Reply:
Just remove this `return` statement, as noted elsewhere.